src/layerPart.cpp

```diff
@@ -65,22 +65,60 @@ void createLayerWithParts(const Settings& settings, SliceLayer& storageLayer, Sl
         result = layer->polygons_.splitIntoParts(union_layers || union_all_remove_holes);
     }
 
-    for (auto& part : result)
+    for (auto& main_part : result)
     {
-        storageLayer.parts.emplace_back();
-        if (part.empty())
+        std::vector<std::pair<SliceLayerPart::WallExposedType, std::vector<SingleShape>>> parts_by_type = {
+            { SliceLayerPart::WallExposedType::BOTTOM_0, bottom_parts.splitIntoParts() },
+            { SliceLayerPart::WallExposedType::TOP, top_parts.difference(bottom_parts).splitIntoParts() },
+            { SliceLayerPart::WallExposedType::SIDE_ONLY, main_part.difference(bottom_parts).difference(top_parts).splitIntoParts() },
+        };
+
+        for (auto& [wall_exposed, parts] : parts_by_type)
         {
-            continue;
+            for (auto& part : parts)
+            {
+                storageLayer.parts.emplace_back();
+                if (part.empty())
+                {
+                    continue;
+                }
+                auto& back_part = storageLayer.parts.back();
+                back_part.wall_exposed = wall_exposed;
+                back_part.outline = part;
+                back_part.boundaryBox.calculate(back_part.outline);
+                if (back_part.outline.empty())
+                {
+                    storageLayer.parts.pop_back();
+                }
+            }
         }
-        storageLayer.parts.back().outline = part;
-        storageLayer.parts.back().boundaryBox.calculate(storageLayer.parts.back().outline);
-        if (storageLayer.parts.back().outline.empty())
+    }
+}
+
+Shape getTopOrBottom(int direction, const std::string& setting_name, size_t layer_nr, const std::vector<SlicerLayer>& slayers, const Settings& settings)
+{
+    auto result = Shape();
+    if (settings.get<size_t>(setting_name) != settings.get<size_t>("wall_line_count") && ! settings.get<bool>("magic_spiralize"))
+    {
+        result = slayers[layer_nr].polygons_;
+        const auto next_layer = layer_nr + direction;
+        if (next_layer >= 0 && next_layer < slayers.size())
         {
-            storageLayer.parts.pop_back();
+            constexpr coord_t EPSILON = 5;
+            const auto wall_line_width = settings.get<coord_t>(layer_nr == 0 ? "wall_line_width_0" : "wall_line_width") - 5;
+            result = result.offset(-wall_line_width).difference(slayers[next_layer].polygons_).offset(wall_line_width);
         }
     }
+    return result;
 }
 
+/*!
+ * \brief Split a layer into parts.
+ * \param settings The settings to get the settings from (whether to union or
+ * not).
+ * \param storageLayer Where to store the parts.
+ * \param layer The layer to split.
+ */
```

Comment:
Shouldn't the documentation be in the header ?

Reply:
Whoops! That's supposed to be the documentation of that other method (not used outside the file), I accidentally placed it on this one.